### DISCUSSION TITLE
Simplify prototyping by adding support for CDNs releases

### DISCUSSION THREAD

**Discussion by gaby** [mantinedev/mantine#2135] [Feedback]
I have been reading the docs for Mantine and can't wait to start using it. One thing I notice compared to other libraries like `MUI` is the lack of docs/support for using Mantine directly from a CDN without installing it. 

Example Docs: https://mui.com/material-ui/getting-started/installation/

Example Usage: https://github.com/mui/material-ui/tree/master/examples/cdn

Unpkg: https://unpkg.com/

The mantine url would look like this (with a umd directory): https://www.unpkg.com/browse/@mantine/core@5.1.6/

Example of how `react` looks: https://www.unpkg.com/browse/react@18.2.0/umd/

I would be very helpfull if Mantine added support for unpkg/minimized cdn releases to ease development/introduction into Mantine.

**Comment by rtivital** [comment]
Hi @gaby, I do not think that UI libraries should be published through cdn. Mantine exports more than 100 components and 50 hooks, if you want all of them in minified umd bundle it will take 2-3mb to load. When you are using modern tools like Next.js or Vite all unused components will be removed at built time (tree shacking).

**Comment by gaby** [comment]
I would argue that MUI is quite big too, and their umd production min file is 494 kB, the development one is 1.72MB

Source: https://unpkg.com/browse/@mui/material@5.10.0/umd/

**Reply by rtivital** [reply]
I was referring to umd bundle sizes without compression, for example `@mantine/core` package alone will be ~450kb – https://bundlephobia.com/package/@mantine/core@5.1.6
Other Mantine packages will add up and do not forger you will also have to install peer dependencies like daysjs, react, react-dom

**Reply by gaby** [reply]
@rtivital Thanks for your help, I will keep using `MUI`.

Related:
- https://reactjs.org/docs/cdn-links.html
- https://www.copycat.dev/blog/reactjs-cdn/
- https://github.com/mui/material-ui/blob/master/examples/cdn/index.html

**Comment by p-himik** [comment]
My 5 c on the topic - it would be much easier to experiment with things and create online reproducible examples in JSFiddle and other services if there was a single URL bundle.

**Reply by rtivital** [reply]
You can experiment in codesandbox

**Reply by p-himik** [reply]
Yeah, I forgot to update the comment - it indeed allows for NPM deps, although it took a bit of time to find it because of a significant amount of "helpful" advice online to "just use unpkg".
